scipy/optimize/tests/test_minpack.py

```diff
@@ -852,6 +853,24 @@ def func(x, a, b, c, d, e):
                       xdata=[1, 2, 3, 4],
                       ydata=[5, 9, 13, 17])
 
+    @pytest.mark.filterwarnings('ignore::RuntimeWarning')
+    def test_gh4555(self):
+        # gh-4555 reported that covariance matrices returned by `leastsq`
+        # can have negative diagonal elements and eigenvalues. (In fact,
+        # they can also be asymmetric.) This shows up in the output of
+        # `scipy.optimize.curve_fit`. Check that it has been resolved.giit
+        def f(x, a, b, c, d, e):
+            return a*np.log(x + 1 + b) + c*np.log(x + 1 + d) + e
+
+        rng = np.random.default_rng(408113519974467917)
+        n = 100
+        x = np.arange(n)
+        y = np.linspace(2, 7, n) + rng.random(n)
+        p, cov = optimize.curve_fit(f, x, y, maxfev=100000)
+        assert np.all(np.diag(cov) > 0)
+        assert np.all(linalg.eigh(cov)[0] > 0)
```

Comment:
Well, there are some failing tests because approximately zero eigenvalues sometimes come out as negative eigenvalues numerically. Let's try this.
```suggestion
        eigs = linalg.eigh(cov)[0]  # separate line for debugging
        assert np.all(eigs > -1e-6)
```